Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Pointers/DefaultPointerMediator.cs

```diff
@@ -167,6 +171,49 @@ public virtual void UpdatePointers()
             {
                 unassignedPointer.IsActive = true;
             }
+
+        }
+
+        private void ApplyCustomPointerBehaviors()
+        {
+            if (CoreServices.InputSystem.FocusProvider is FocusProvider focusProvider)
```

Comment:
It feels weird to me that the default pointer mediator, which is a thing that the focus provider uses, has to then reach all the way back out of the system and find the focus provider (and treat it directly as THE default focus provider).

I think that doing things this way avoid some breaking changes, but also still adds code cruft (i.e. generally speaking casting to specific implementations from interfaces is not really the right thing to do).

Reply:
More specifically maybe it means that both:

1) The focus provider should be provided to the mediator in its constructor (or in some initialization step or in the function call itself)
2) The GetPointerBehavior shouldn't depend on the specific implementation of the focus provider - it may not be on the focus provider interface (maybe it's on should be on some IPointerBehaviorProvider interface or something)

Reply:
> I think that doing things this way avoid some breaking changes, but also still adds code cruft (i.e. generally speaking casting to specific implementations from interfaces is not really the right thing to do).

I thought about adding an interface, but wanted to avoid adding more interfaces. Additionally, this is a behavior specific to the input system / focus provider / mediator combo that the default MRTK uses. For example, a different system could provide this functionality in a different (simpler) way.

> It feels weird to me that the default pointer mediator, which is a thing that the focus provider uses, has to then reach all the way back out of the system and find the focus provider (and treat it directly as THE default focus provider).

I agree. I initially had the behavior set per mediator. Then I ran into the following problem: the behaviors would reset whenever I dropped and raised hands. This is because the mediator is created _per input source_. It seemed like the simplest solution was to specify the behaviors in a location that was not going to change per source, which is why I picked the focus provider.

I could try to instead have the provider specify the pointerbehavior per mediator every time the mediator is instantiated, I suppose...for some reason just grabbing the focusprovider behavior actually seems like "simpler code" to me tbh.

Reply:
We chatted. I will try the following approach, though if run into major issues, will probably move off of it since it can be done at later time:

```
interface IPointerPreferences
{
  // Returns the desired behavior for a given pointer
  public PointerBehavior GetPointerBehavior(IMixedRealityPointer ptr);
}
```

Then I would modify DefaultPointerMediator to take as a constructor:

```
public DefaultPointerMediator(IPointerPreferences pointerPreferences)
{
  this.pointerPreferences = pointerPreferences;
}
```

Then instead of accessing `CoreServices.InputSystem.FocusProvider` I would just look at `pointerPreferences`.